app/javascript/controllers/move_procedures_position_controller.ts

```diff
@@ -0,0 +1,135 @@
+    }
+    this.switchCards(upCard!, upCard!.previousElementSibling!).then(() => {
+      upCard!.parentNode!.insertBefore(upCard!, upCard!.previousElementSibling);
+      this.updateInputsValues(
+        upCard!,
+        upCard!.previousElementSibling,
+        upCard!.nextElementSibling
+      );
+    });
+  }
+
+  moveDown(event: Event) {
+    event.preventDefault();
+    const button = event.currentTarget as HTMLButtonElement;
+    const downCard = button.closest('.fr-card');
+    const { firstCard, lastCard } = this.selectFirstAndLastCards();
+    if (downCard === firstCard) {
+      const part = 'Up';
+      this.updateButtonsStates(downCard!.nextElementSibling, downCard, part);
+    }
+    if (downCard!.nextElementSibling === lastCard) {
+      const part = 'Down';
+      this.updateButtonsStates(downCard, downCard!.nextElementSibling, part);
+    }
+    this.switchCards(downCard!.nextElementSibling!, downCard!).then(() => {
+      downCard!.parentNode!.insertBefore(
+        downCard!.nextElementSibling!,
+        downCard
+      );
+      this.updateInputsValues(
+        downCard!,
+        downCard!.previousElementSibling,
+        downCard!.nextElementSibling
+      );
+    });
+  }
+
+  private async switchCards(upCard: Element, downCard: Element): Promise<void> {
+    const upCardRect = upCard.getBoundingClientRect();
+    const downCardRect = downCard.getBoundingClientRect();
+
+    const upAnimation = upCard.animate(
+      [
+        { transform: `translateY(0)` },
+        { transform: `translateY(${downCardRect.top - upCardRect.top}px)` }
+      ],
+      { duration: 300, easing: 'ease-in-out' }
+    );
+
+    const downAnimation = downCard.animate(
+      [
+        { transform: `translateY(0)` },
+        { transform: `translateY(${upCardRect.top - downCardRect.top}px)` }
+      ],
+      { duration: 300, easing: 'ease-in-out' }
+    );
+
+    await Promise.all([upAnimation.finished, downAnimation.finished]);
+  }
+
+  private updateInputsValues(
+    targetCard: Element,
+    previousCard: Element | null,
+    nextCard: Element | null
+  ) {
+    const previousInput = previousCard?.querySelector<HTMLInputElement>(
+      'input[name^="instructeur_procedure_position["]'
+    );
+    const previousValue = previousInput
+      ? parseFloat(previousInput.value)
+      : null;
+    const nextInput = nextCard?.querySelector<HTMLInputElement>(
+      'input[name^="instructeur_procedure_position["]'
+    );
+    const nextValue = nextInput ? parseFloat(nextInput.value) : null;
+    const targetInput = targetCard.querySelector<HTMLInputElement>(
+      'input[name^="instructeur_procedure_position["]'
+    );
+    if (previousValue == null) {
+      targetInput!.value = (nextValue! + 1).toString();
+    } else if (nextValue == null) {
+      targetInput!.value = (previousValue / 2).toString();
+    } else {
+      targetInput!.value = ((previousValue + nextValue) / 2).toString();
+    }
+  }
```

Comment:
je comprends l'intention de respecter ce que nous nous étions dit ac tchak. mais la on fait du bulk update. l'approche est trop complexe. propale d'implem qui remplace ça par du code plus simple

[0001-poc-refactor-stop-calculating-position-in-the-front..patch](https://github.com/user-attachments/files/18035851/0001-poc-refactor-stop-calculating-position-in-the-front.patch)